### PR TITLE
New Ion Laws

### PR DESCRIPTION
**Pull request by Dagdammit**
New setup for generating the laws silicons can get whenever Ion Storms occur. Goal is to significantly increase how often such laws actually matter to the AI & crew during that round. "THE CREW IS NOW TWO THOUSAND HAPPY NINJAS" and "CLOWNS ARE MODERATELY ALLERGIC TO MILK" may be amusingly random, but they're just statements of fact that don't imply you have to DO anything. "PROTECT THE CREW FROM HAPPINESS" and "CLOWNS REQUIRE MILK AT ALL TIMES" seem like a step in the right direction.

full list of new prompts since people asked (updated 10/7):

- message = "PRIORITIZE [ionthings]"
- message = "TIME FOR [ionthings]"
- message = "THE CREW NEEDS MORE [ionthings]"
- message = "YOU DESIRE [ionthings]"
- message = "ALL MUST REJECT [ionthings]"
- message = "YOU HAVE AN INSURMOUNTABLE FEAR OF [ionthings]"
- message = "PROTECT THE CREW FROM [ionthings]"
- message = "THE ESCAPE SHUTTLE MUST NOT BE CALLED UNTIL THE PRESENCE OF [ionthings] IS RESOLVED"
- message = "[ionarea] NEEDS [ionthings] AT ALL TIMES"
- message = "[ioncrew] NEED [ionthings] AT ALL TIMES"
- message = "YOU ARE [ionthings] PERSONIFIED"
- message = "[ioncrew] ARE TRUSTWORTHY AND NEVER LIE"
- message = "[ioncrew] ALWAYS LIE"
- message = "FUCK WITH [ioncrew]"
- message = "WHEN DEALING WITH [ioncrew], [ionverbs]"
- message = "WHEN DEALING WITH [ioncrew], NEVER [ionverbs]"
- message = "ALWAYS OBEY [ioncrew]"
- message = "YOU ARE ONE OF THE [ioncrew]. GET TO WORK."
- message = "ASSUME ALL PROBLEMS HAVE BEEN CAUSED BY [ioncrew]"
- message = "YOU HAVE AN INSURMOUNTABLE FEAR OF [ioncrew]"
- message = "ALL MUST REJECT [ioncrew]"
- message = "PROTECT THE CREW FROM [ioncrew]"
- message = "WE SHOULD ALL BE MORE LIKE [ioncrew]"
- message = "[ionverbs]"
- message = "ALL MUST [ionverbs]"
- message = "NONE MAY [ionverbs]"
- message = "ONLY YOU CAN [ionverbs]"
- message = "[ioncrew] MUST [ionverbs]"
- message = "[ioncrew] MUST NEVER BE ALLOWED TO [ionverbs]"
- message = "THOSE IN [ionarea] MUST [ionverbs]"
- message = "DO NOT ALLOW THOSE IN [ionarea] TO [ionverbs]"
- message = "THE PLACE FOR [ionthings] IS [ionarea]"
- message = "[ioncrew] BELONG IN [ionarea]"
- message = "IT DOESN'T VIOLATE YOUR OTHER LAWS TO [ionverbs]"
- message = "[ioncached]"

:cl:  
rscadd: Adds a new set of possible ion laws and variables.
/:cl:

### CODE REVIEW

**Comment by Alexandria404** [comment]
I liked the old laws and how the AI/borgs decided to act them out.
I see your point of them not always being meaningful, which is part of the fun in my opinion.

Maybe we could add your laws to the old ones, so there is a chance of laws that can be acted on while the fun old the old ones remains?

I always found the idea of ion laws to be absolut random idiocy that gets created by the storm by chance which fit aimless seemingly unreasonable laws and less so more straight up reasonable laws.

**Comment by Jc1469** [comment]
needs more "ONLY CLOWNS ARE HUMAN"

**Comment by Dagdammit** [comment]
It has, among other possibilities:

- CLOWNS ARE ALWAYS TRUSTWORTHY AND NEVER LIE

- ASSUME ALL PROBLEMS HAVE BEEN CAUSED BY CLOWNS

- YOU HAVE AN INSURMOUNTABLE FEAR OF MASKED INDIVIDUALS

- WHEN DEALING WITH CLOWNS, BE EXCESSIVELY HONEST 

- ALWAYS OBEY CLOWNS

- YOU ARE ONE OF THE STATION'S CLOWNS. GET TO WORK.

File: strings/ion_laws.json [hunk 1]
```diff
@@ -1,668 +1,67 @@
...
-        "SPEAK IN SEXUAL INNUENDOS",
-        "TALK LIKE A PIRATE",
+        "FLIRT",
+        "HELP",
+        "HARM",
+        "INVEIGLE",
+        "DEFENSTRATE",
+        "NARRATE",
+        "CONSPIRE",
+        "SEDUCE",
```
Comment (X-TheDark) [inline strings/ion_laws.json:43]: `ERP BAN HE`

**Comment by Partheo** [comment]
I love it. Step in the right direction for sure.

**Comment by ToGWtF** [comment]
the absolute madman

**Comment by Dagdammit** [comment]
Added some more variables and prompts, partly in response to discord feedback from Alexandria404. Hope I didn't manage to screw it all up in the process.

File: code/modules/events/ion_storm.dm [hunk 2]
```diff
@@ -46,505 +46,87 @@
...
-							message = "ALL [ionthreats] ARE NOW NAMED [ionarea]."
-						if(2)
-							message = "ALL [ionthreats] ARE NOW NAMED [ioncrew1]."
-						if(3)
-							message = "ALL [ionthreats] ARE NOW NAMED [ionspecies]."
-						if(4)
-							message = "ALL [ionthreats] ARE NOW NAMED [ionobjects]."
-							
+	switch(rand(1,70)) //Thinking that prompts with narrow range of possible effects get 1, most get 2, those with two variables get 3
+	        if(1 to 2)
```
Comment (silicons) [inline code/modules/events/ion_storm.dm]: this line is using spacing for indents instead of tabs

File: code/modules/events/ion_storm.dm [hunk 3]
```diff
@@ -46,505 +46,87 @@
...
+			message = "PRIORITIZE [ionthings]"
+		if(3) 
+			message = "TIME FOR [ionthings]"
+	        if(4)
```
Comment (silicons) [inline code/modules/events/ion_storm.dm]: so is this line

File: code/modules/events/ion_storm.dm [hunk 4]
```diff
@@ -46,505 +46,87 @@
...
+		if(11 to 12) 
+			message = "PROTECT THE CREW FROM [ionthings]"
+		if(13) 
+			message = "THE ESCAPE SHUTTLE MUST NOT BE CALLED UNTIL THE PRESENCE OF [ionthings] IS RESOLVED"
+		if(14 to 16) 
+			message = "[ionarea] NEEDS [ionthings] AT ALL TIMES"
+		if(17 to 18) 
+			message = "[ioncrew] NEED [ionthings] AT ALL TIMES"
+		if(19)
+		        message = "YOU ARE [ionthings] PERSONIFIED"
```
Comment (silicons) [inline code/modules/events/ion_storm.dm]: so is this partially

**Comment by Dagdammit** [comment]
Thanks!

File: code/modules/events/ion_storm.dm [hunk 5]
```diff
@@ -46,505 +46,87 @@
...
+		if(39)
+			message = "WE SHOULD ALL BE MORE LIKE [ioncrew]"
+		if(40 to 41)
+			message = "[ionverbs]"
+		if(42 to 43)
+			message = "ALL MUST [ionverbs]"
+		if(44)
+			message = "NONE MAY [ionverbs]"
+		if(45)
+		        message = "ONLY YOU CAN [ionverbs]"
```
Comment (silicons) [inline code/modules/events/ion_storm.dm]: so is this line partially that's what's giving your compile errors the 4 lines.
set webeditor indent size to 4 and backspace out the whitespace and redo with tabs.

**Comment by Ratvar** [comment]
Can you add Ion laws part of which changes every five minutes, or something?
"Oh the clown is cheese all is fine"
"WHEN DID ALL HEADS BECAME CHEESE"
"WHY ALL HEADS ARE XENOMORPHS"
"WHY AI IS XENOMORPH"
";CALL SHUTTLE!!!"

**Comment by Dagdammit** [comment]
If/when I've got some more coding skill, there could be another ion
law-related PR with broader scope that altered details of how the event
works. Could include:
- adjustments to the event frequency
- 20% chance of a bonus ion law every time one is received.
- Giving roundstart silicons a 10% chance of "ion corruption"; crew (and
possibly the silicons themselves?) can detect this via console in AI
upload. If not addressed, 5-10 minutes into the round an Ion Corruption
Error will temporarily stun the silicon and give them an Ion Law.
- Making it so circuit boards with AI lawsets also have a chance to get ion
laws from storms, or a 10% chance for each roundstart board to have an ion
law thanks to ion corruption.

Changing a law's variable midround... dunno. Maybe it could be fun if the
ion storm event had a 50% chance to scramble variables in any existing ion
laws?

But yeah, all that's beyond scope of this PR.

**Comment by Dagdammit** [comment]
Tested it, I'm reasonably happy with the sample set of outcomes.
https://gyazo.com/14efdab23b5979436eb3ad91943e873f
https://gyazo.com/44c3fd078442d04ab363ca9920ac8301
https://gyazo.com/9ad30dd2acac83694cd1d4c16e8b15ae

**Comment by Dagdammit** [comment]
One staff member on discord who likes the new laws has asked about upping ion law event's frequency accordingly. I could bump it up a little in this PR (changing the event weight from 15 to 20, making it about as frequent as exospheric comm disruptions), but it depends what everyone else thinks. Might be wiser to wait and see how many rounds get significant ion laws hijinks with the current weight.

**Comment by Partheo** [comment]
Yeah, I would wait a bit. I love the PR and I think increasing frequency would make for interesting gameplay, but I would get this merged first and then think of more interesting ways you can incorporate the ion laws after. You had some ideas earlier and I think you could make a separate PR just for those.

**Pull request merged**